### ISSUE TITLE
Assembly version should step when the effective interface changes

### ISSUE THREAD

**Issue by SteveGilham** [jbevain/cecil#502]
The root cause of the problem I reported as issue #500 is that the effective interface (which includes the internal methods used by the associated assemblies) of the Mono.Cecil assembly changed between beta7 and final, but the assembly version, which is used by the runtime as a shorthand for the effective interface, did not, and there was no automated check in place to warn of this.  Interface versioning, even with a strict Open/Closed approach, is a tricky problem

It's a bit late now that 0.10.0 has been released, with AssemblyVersion("0.10.0.0") rather than some AssemblyVersion("0.10.0.xxxxx") where xxxxx is greater than any similar value built into any earlier public beta, but such a check could be instituted now, so that future releases can be spared such compatibility issues.

**Comment by SimonCropp** [comment]
From my perspective this was a problem with nunits choice of using an unstable in a non-isolated appdomain. no changes should be required by mono cecil. 

@jbevain so i think this should be closed

**Comment by SteveGilham** [comment]
Yes, it's an AppDomain-caused issue that makes interfaces even wider than you think.
It can go away when everything is on .net 5 and later, and the root cause is removed.
